pkg/controller/pxc/replication.go

```diff
@@ -55,7 +55,7 @@ func (r *ReconcilePerconaXtraDBCluster) ensurePxcPodServices(cr *api.PerconaXtra
 			return errors.Wrap(err, "failed to set owner to external service")
 		}
 
-		err = r.createOrUpdate(cr, svc)
+		err = r.createOrUpdateService(cr, svc, len(cr.Spec.PXC.Expose.Annotations) == 0)
```

Comment:
Don't delete annotations. 
STR:
Add annotations to ignore
Add annotation to service (the same behavior for haproxy/proxysql) -> OK
   ```
 serviceAnnotations:
      service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
```
Only annotations from ignore + annotations from service! 
Than I try to delete service annotations. And see that they were not deleted. 
```
Annotations:              kubernetes.civo.com/cluster-id: 5b4600cd-34c1-4ff5-88e8-4f6ed4198e5e
                          kubernetes.civo.com/firewall-id: ccd93a19-3428-4055-8796-bdd13cfa0c35
                          kubernetes.civo.com/loadbalancer-algorithm: round_robin
                          kubernetes.civo.com/loadbalancer-id: d688ee14-b0ed-45a9-94fe-66c8660c539a
                          kubernetes.civo.com/loadbalancer-name: laboriosam_percona_cluster-pxc-cluster1-proxysql
                          percona.com/last-config-hash:
                            eyJwb3J0cyI6W3sibmFtZSI6Im15c3FsIiwicG9ydCI6MzMwNiwidGFyZ2V0UG9ydCI6MH0seyJuYW1lIjoibXlzcWwtYWRtaW4iLCJwb3J0IjozMzA2MiwidGFyZ2V0UG9ydCI6MH...
                          service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
```
Logic for annotations should be also implement on labels.

Reply:
@nmarukovich it is by design